server/graphql-context.js

```diff
@@ -1,16 +1,17 @@
+const AtLoader = require('./models/loaders/AtLoader');
+const BrowserLoader = require('./models/loaders/BrowserLoader');
+
 const getGraphQLContext = ({ req }) => {
     const user =
         req && req.session && req.session.user ? req.session.user : null;
 
-    let transaction;
-    if (req) {
-        transaction = req.transaction;
-    } else {
-        // Req will not be defined when queries are made with apolloServer.executeOperation()
-        transaction = false;
-    }
+    // Req will not be defined when queries are made with apolloServer.executeOperation()
+    let transaction = req ? req.transaction : false;
+
+    const atLoader = AtLoader();
+    const browserLoader = BrowserLoader();
 
-    return { user, transaction };
+    return { user, atLoader, browserLoader, transaction };
```

Comment:
Isn't this going back to how this file used to be? Should it? Didn't this have some implications to automation?

Reply:
Thanks for asking about this. This is definitely one of the larger changes in the PR. So you're right that the automation PR changed the way the atLoader and browserLoader worked. However that change should be considered more of a refactor or a simplification than something strictly required for automation. When I reviewed the first major automation PR I had a chance to go over that change in detail, including discussing with Stalgia's her motivation for making the change. 

When she introduced the "singleton" architecture which is currently still in use on main, she had not realized that it was possible to dynamically add new AT and browser versions. Therefore, in the first version of the PR, the caching inherent to the singleton approach would result in stale data being loaded for AT and Browser versions, until the server was restarted. To fix this she added several functions for clearing the cache. This was a good solution at the time to quickly unblock the PR, but a I think a couple of developments since then made going back to the old approach preferable again.

Number one, one of the big benefits of using a singleton architecture was the fact that passing context to all the resolvers was no longer needed. This is no longer the case. The context is now a required field, because the transaction is a required field.

Number two, the other big benefit of using a singleton architecture was the fact that the cache could be more aggressive and hit the database less often, almost never. In the previous approach we would have had to query the database once during every request. However this aggressive caching was the real show stopper when I came along to test this feature. The issue is that the caching would not respect the current transaction. And in addition to possibly causing issues in production for code that mutates the AT and browser versions before the transaction is committed, the main problem is that this actually made testing with a long-lived transaction completely impossible.

Hopefully that "little" essay is enough to understand the reasoning for the change. Thanks for your patience as I try to explain that.

Reply:
Makes sense! Haven't noticed any significant setbacks with this change during my perusal of the app with this PR.

> Hopefully that "little" essay is enough to understand the reasoning for the change. Thanks for your patience as I try to explain that.

It does! Thanks for sharing this "essay" :), and I appreciate the work and discussions you and @stalgiag must have to had to ensure continuous changes to this file are as safe as possible to make.